api/src/pcapi/routes/native/v1/bookings.py

```diff
@@ -196,7 +196,8 @@ def is_ended_booking(booking: Booking) -> bool:
         return False
 
     if (booking.stock.canHaveActivationCodes and booking.activationCode) or (
-        booking.stock.offer.subcategoryId in FREE_OFFER_SUBCATEGORIES_TO_ARCHIVE and booking.stock.price == 0
+        booking.stock.offer.subcategoryId in [subcategory.id for subcategory in FREE_OFFER_SUBCATEGORIES_TO_ARCHIVE]
```

Comment:
**Suggestion :** 
Plutôt que reconstruire ce tableau à chaque exécution, et comme on n'utilise que l'`id` à chaque utilisation de `FREE_OFFER_SUBCATEGORIES_TO_ARCHIVE`, ne serait-il pas plus judicieux de définir directement une liste d'ids, en renommant en `FREE_OFFER_SUBCATEGORY_IDS_TO_ARCHIVE` ?
```
FREE_OFFER_SUBCATEGORY_IDS_TO_ARCHIVE = [
    subcategories.ABO_MUSEE.id,
    subcategories.CARTE_MUSEE.id,
    subcategories.ABO_BIBLIOTHEQUE.id,
    subcategories.ABO_MEDIATHEQUE.id,
]
```
Même si le gain de performance est peut-être anecdotique, ça a le mérite de simplifier aussi le code.

Reply:
J'ai du arbitrer dans la PR précédente, au début j'avais fait une liste d'id, mais je me suis dit que on perdait de l'info dans la liste, donc j'ai fais l'enum.

En revanche, pour la simplification du code, c'est vrai que ça rends plus complexe, mais en prenant en compte l'arbitrage précédent, je peux : 

- laisser comme ça, après tout on a ces for in partout dans le code et c'est donc lisible pour les devs
- faire une autre constante pour fabriquer la liste d'id depuis la 1er

le 2nd point me parait un peu anticipé, au final je préfère comme c'est.